tests/copp/test_copp.py

```diff
@@ -207,80 +203,96 @@ def _gather_test_params(tbinfo, duthost, request):
             peerip = bgp_peer["peer_addr"]
             break
 
-    logging.info("nn_target_port {} nn_target_interface {}".format(nn_target_port, nn_target_interface))
+    nn_target_namespace = mg_facts["minigraph_neighbors"][nn_target_interface]['namespace']
+
+    logging.info("nn_target_port {} nn_target_interface {} nn_target_namespace {}".format(nn_target_port, nn_target_interface, nn_target_namespace))
 
     return _COPPTestParameters(nn_target_port=nn_target_port,
                                swap_syncd=swap_syncd,
                                topo=topo,
                                myip=myip,
                                peerip = peerip,
-                               nn_target_interface=nn_target_interface)
+                               nn_target_interface=nn_target_interface,
+                               nn_target_namespace=nn_target_namespace)
 
-def _setup_testbed(dut, creds, ptf, test_params):
+def _setup_testbed(dut, creds, ptf, test_params, tbinfo):
     """
         Sets up the testbed to run the COPP tests.
     """
+    if dut.is_multi_asic:
+       logging.info("Adding iptables rules and enabling eth0 port forwarding")
+       http_proxy, https_proxy = copp_utils._get_http_and_https_proxy_ip(creds)
+       # IP Table rule for http and ptf nn_agent traffic.
+       dut.command("sudo sysctl net.ipv4.conf.eth0.forwarding=1")
+       mgmt_ip = dut.host.options["inventory_manager"].get_host(dut.hostname).vars["ansible_host"]
+       # Rule to communicate to http/s proxy from namespace
+       dut.command("sudo iptables -t nat -A POSTROUTING -p tcp --dport 8080 -j SNAT --to-source {}".format(mgmt_ip))
+       dut.command("sudo ip -n {} rule add from all to {} pref 1 lookup default".format(test_params.nn_target_namespace, http_proxy))
+       if http_proxy != https_proxy:
+           dut.command("sudo ip -n {} rule add from all to {} pref 2 lookup default".format(test_params.nn_target_namespace, https_proxy))
+       # Rule to communicate to ptf nn agent client from namespace
+       ns_ip = dut.shell("sudo ip -n {} -4 -o addr show eth0".format(test_params.nn_target_namespace) + " | awk '{print $4}' | cut -d'/' -f1")["stdout"]
+       dut.command("sudo iptables -t nat -A PREROUTING -p tcp --dport 10900 -j DNAT --to-destination {}".format(ns_ip))
+       dut.command("sudo ip -n {} rule add from {} to {} pref 3 lookup default".format(test_params.nn_target_namespace, ns_ip, tbinfo["ptf_ip"]))
```

Comment:
For readability it would be good if this was a separate helper method like "setup_multi_asic_proxy" or something similar?

Reply:
@daall Updated.